packages/flutter/lib/src/services/system_chrome.dart

```diff
@@ -81,6 +81,67 @@ enum SystemUiOverlay {
   bottom,
 }
 
+/// Signature for listening to changes in the [SystemUiMode].
+///
+/// Used by [SystemChrome.setEnabledSystemUIMode].
+typedef SystemUiChangeCallback = void Function(bool systemOverlaysAreVisible);
+
+/// Specifies Android fullscreen mode.
+///
+/// Used by [SystemChrome.setEnabledSystemUIMode].
+enum SystemUiMode {
+  /// Fullscreen display without status or navigation bars.
```

Comment:
It may be more useful if the first line summary already hints at how the modes are different (right now  the first three modes share the same summary)